### DISCUSSION TITLE
How to properly skip chromes download prompt in incognito?

### DISCUSSION THREAD

**Discussion by DIFSRIP** [webdriverio/webdriverio#12191] [Q&A]
Hey WDIO Community,

I've encountered an issue while using WebDriverIO where I'm unable to skip Chrome's download prompt when running tests in incognito mode. Interestingly, the same tests run smoothly in non-incognito mode. I've attempted to resolve this by configuring the Chrome capabilities with the following options:

```javascript
capabilities: [
  {
    browserName: 'chrome',
    'goog:chromeOptions': {
      prefs: {
        download: {
          default_directory: '/path/to/your/known/folder',
          prompt_for_download: false,
          directory_upgrade: true,
        },
      },
    },
  },
],
```

Despite these configurations, the download prompt still persists in incognito mode. Has anyone else faced a similar challenge, and if so, do you have any suggestions or workarounds to bypass Chrome's download prompt in WDIO when in incognito mode using latest chromedriver?

I also found this related post which did not help: https://support.google.com/chrome/thread/238034785/unable-to-disable-the-download-prompt-for-incognito-window-while-running-automation-in-chrome?hl=en

Looking forward to your insights and solutions!

Best regards,
Ben

**Comment by christian-bromann** [comment]
Can you try to run this example: https://github.com/webdriverio/example-recipes/tree/84dda93011234d0b2a34ee0cfb3cdfa2a06136a5/testDownloadBehavior ?

> the download prompt still persists in incognito mode

What makes your automated session run in incognito mode? Is it the `directory_upgrade` flag?

This seems like a very Chrome specific issue, I am not sure if WebdriverIO can fix this.

**Reply by Templada** [reply]
Hi Christian,

we are using different profiles which are inheriting from each other. The profile we are inheriting from already has set the ` "--incognito"` flag.

I am as well pretty sure that this is a chrome-driver issue, since also others reporting about the issue. Seems to be around mid/end 2023.

Will check and open an issue on chromium tomorrow. Keep you posted here.

**Reply by Templada** [reply]
As promissed here the chromium [issue](https://issues.chromium.org/issues/324301023).

**Reply by DIFSRIP** [reply]
Hi @christian-bromann thanks for the quick reply. 
As @Templada said, we have set `--incognito` in our chrome args.
We have done another test and it seems that it is consistently failing (not skipping the dialog) if executed in non-headless or incognito. Headless is working, no matter if incognito or not is set.
